nats.go

```diff
@@ -1283,38 +1283,47 @@ func (nc *Conn) ConnectedUrl() string {
 	if nc == nil {
 		return _EMPTY_
 	}
-	nc.mu.Lock()
-	defer nc.mu.Unlock()
```

Comment:
Defers can be expensive but in non performance paths where the code between locks is not drop dead simple I prefer them for ease of reading the code later.